### PR TITLE
DOC: special: improve the docstrings for `gamma` and `gammasgn`

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

None

#### What does this implement/fix?

This completes the docstrings for `gamma` and `gammasgn`.

#### Additional information

For `gamma`:

- Add a reference
- Tweak the wording to be more standard

For `gammasgn`:

- Add a definition
- Add parameters/returns sections
- Add a reference
- Add examples

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -2693,22 +2693,36 @@ def add_newdoc(name, doc):
...
-        The value(s) of gamma(z)
+    scalar or ndarray
+        Values of the Gamma function
+
+    Notes
+    -----
+    The gamma function is often referred to as the generalized
+    factorial since :math:`\Gamma(n + 1) = n!` for natural numbers
+    :math:`n`. It also satisfies the recurrence relation
+    :math:`\Gamma(z + 1) = z \cdot \Gamma(z)`.
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Minor: maybe explicitly state that the recurrence holds for real z, hence for natural values z=n, it becomes m!

**Review by ev-br — APPROVED** [review]
One minor suggestion. Looks great with it or without

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (person142) [reply scipy/special/add_newdocs.py]: Updated the language in https://github.com/scipy/scipy/pull/10665/commits/af706ebbb0048fa6d337703935282b4b03f6d143, what do you think?

Comment (ev-br) [reply scipy/special/add_newdocs.py]: Great

**Comment by ev-br** [comment]
-dbg CI failure does not seem related, so let's merge it to keep the ball rolling. Thanks Joah

**Pull request merged**